src/azure-cli/azure/cli/command_modules/cosmosdb/_validators.py

```diff
@@ -61,3 +62,116 @@ def validate_virtual_network_rules(ns):
         for item in ns.virtual_network_rules:
             virtual_network_rules_list.append(VirtualNetworkRule(id=item))
         ns.virtual_network_rules = virtual_network_rules_list
+
+
+def validate_role_definition_body(cmd, ns):
+    """ Extracts role definition body """
+    from azure.mgmt.cosmosdb.models import Permission
+    from azure.cli.core.util import get_file_json, shell_safe_json_parse
+    import os
+    if ns.role_definition_body is not None:
+        if os.path.exists(ns.role_definition_body):
+            role_definition = get_file_json(ns.role_definition_body)
+        else:
+            role_definition = shell_safe_json_parse(ns.role_definition_body)
+
+        if not isinstance(role_definition, dict):
+            raise CLIError('Invalid role defintion. A valid dictionary JSON representation is expected.')
```

Comment:
Use right error type in https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/azclierror.py